test/gleam/binary_test.gleam

```diff
@@ -0,0 +1,41 @@
+import gleam/binary
+import gleam/should
+
+pub fn length_test() {
+  binary.byte_size(binary.from_string("hello"))
+  |> should.equal(5)
+
+  binary.byte_size(binary.from_string(""))
+  |> should.equal(0)
+}
+
+pub fn part_test() {
+  binary.from_string("hello")
+  |> binary.part(0, 5)
+  |> should.equal(binary.from_string("hello"))
+
+  binary.from_string("hello")
+  |> binary.part(0, 0)
+  |> should.equal(binary.from_string(""))
+
+  binary.from_string("hello")
+  |> binary.part(2, 2)
+  |> should.equal(binary.from_string("ll"))
+
+  binary.from_string("hello")
+  |> binary.part(5, -2)
+  |> should.equal(binary.from_string("lo"))
```

Comment:
What does this do when the index is out of bounds and when the length is too long?

Reply:
I think it's badarg. I'm not sure if we want it to return a result or if `binary.part(binary.from_string("12345", 3, 100)` should return "45".

Reply:
All functions in the standard library should be total, let's add some tests for these cases.

Reply:
Result or discard extra if length/position too high?

Reply:
I'm leaning towards result. because with assert/try it's easy to handle the case where it doesn't split cleanly. However if you return a string shorter than the length specified it's easier to forget to do the appropriate checks. Also a result matches the semantics of the erlang binary.part function

Reply:
Result sounds good to me :)